tidy3d/components/mode/mode_solver.py

```diff
@@ -353,8 +392,22 @@ def _get_data_with_group_index(self) -> ModeSolverData:
     @cached_property
     def grid_snapped(self) -> Grid:
         """The solver grid snapped to the plane normal and to simulation 0-sized dims if any."""
-        grid_snapped = self._solver_grid.snap_to_box_zero_dim(self.plane)
-        return self.simulation._snap_zero_dim(grid_snapped)
+        return self._grid_snapped(simulation=self.simulation, plane=self.plane)
+
+    @classmethod
+    def _grid_snapped(cls, simulation: Simulation, plane: Box) -> Grid:
+        """The solver grid snapped to the plane normal and to simulation 0-sized dims if any."""
+        solver_grid = cls._get_solver_grid(
+            simulation=simulation,
+            plane=plane,
+            keep_additional_layers=False,
+            truncate_symmetry=True,
+        )
+        # snap to plane center along normal direction
```

Comment:
@momchil-flex @dbochkov-flexcompute  fyi

I tracked down the issue to this PR:
https://github.com/flexcompute/tidy3d/pull/851

It's pretty minor but it comes up when the simulation only has one grid cell in a direction and has a different center than the mode plane. Before that PR, the normal dim was snapped to mode plane and 0 size tangential dims were snapped to simulation. After that PR, the normal dim was snapped to mode plane and all 0 size dims (including normal dim) were snapped to simulation.